cmd/discovery-service/main.go

```diff
@@ -121,17 +122,22 @@ func run(ctx context.Context, logger *zap.Logger) error {
 
 	recoveryOpt := grpc_recovery.WithRecoveryHandler(recoveryHandler(logger))
 
+	limiter := limits.NewIPRateLimiter(1, limits.RequestsPerSecondMax)
+	limiter.IPAddrGarbageCollector()
```

Comment:
this would block here, as we're not starting this is a goroutine. probably makes sense to start the GC in the errgroup `eg` below